File: ruby/v2.1/create_floodlight_activity_group.rb
#!/usr/bin/env ruby
# Encoding: utf-8
#
# Copyright:: Copyright 2015, Google Inc. All Rights Reserved.
#
# License:: Licensed under the Apache License, Version 2.0 (the "License");
#           you may not use this file except in compliance with the License.
#           You may obtain a copy of the License at
#
#           http://www.apache.org/licenses/LICENSE-2.0
#
#           Unless required by applicable law or agreed to in writing, software
#           distributed under the License is distributed on an "AS IS" BASIS,
#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
#           implied.
#           See the License for the specific language governing permissions and
#           limitations under the License.
#
# This example creates a new activity group for a floodlight configuration.
#
# To get a floodlight configuration ID, run get_advertisers.rb.

require_relative 'dfareporting_utils'

def create_floodlight_activity_group(dfareporting, args)
  # Create a new floodlight activity group resource to insert
  activity_group = {
    :floodlightConfigurationId => args[:floodlight_config_id],
    :name => 'Example Floodlight Activity Group',
    :type => 'COUNTER'
  }

  # Insert the floodlight activity group
  result = dfareporting.floodlight_activity_groups.insert(
    :profileId => args[:profile_id]
  ).body(activity_group).execute()

  # Display results
  puts 'Created floodlight activity group with ID %d and name "%s".' %
      [result.data.id, result.data.name]
end

if __FILE__ == $0
  # Retrieve command line arguments
  args = DfaReportingUtils.get_arguments(
      ARGV, :profile_id, :floodlight_config_id)

  # Authenticate and initialize API service
  dfareporting = DfaReportingUtils.setup()

  create_floodlight_activity_group(dfareporting, args)
end

File: python/v2.3/create_floodlight_activity_group.py
#!/usr/bin/python
#
# Copyright 2015 Google Inc. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""This example creates a new activity group for a floodlight configuration.

To get a floodlight configuration ID, run get_advertisers.py.
"""

import argparse
import sys

import dfareporting_utils
from oauth2client import client

# Declare command-line flags.
argparser = argparse.ArgumentParser(add_help=False)
argparser.add_argument(
    'profile_id', type=int,
    help='The ID of the profile to add a placement for')
argparser.add_argument(
    'floodlight_config_id', type=int,
    help='The ID of the floodlight config to create a group for')


def main(argv):
  # Retrieve command line arguments.
  flags = dfareporting_utils.get_arguments(argv, __doc__, parents=[argparser])

  # Authenticate and construct service.
  service = dfareporting_utils.setup(flags)

  profile_id = flags.profile_id
  floodlight_config_id = flags.floodlight_config_id

  try:
    # Construct and save floodlight activity group.
    activity_group = {
        'name': 'Test Floodlight Activity Group',
        'floodlightConfigurationId': floodlight_config_id,
        'type': 'COUNTER'
    }

    request = service.floodlightActivityGroups().insert(
        profileId=profile_id, body=activity_group)

    # Execute request and print response.
    response = request.execute()

    print ('Created floodlight activity group with ID %s and name "%s".'
           % (response['id'], response['name']))

  except client.AccessTokenRefreshError:
    print ('The credentials have been revoked or expired, please re-run the '
           'application to re-authorize')


if __name__ == '__main__':
  main(sys.argv)
